pkg/proc/native/proc_linux.go

```diff
@@ -330,14 +340,19 @@ const (
 	trapWaitDontCallExitGuard
 )
 
-func (dbp *nativeProcess) trapWaitInternal(pid int, options trapWaitOptions) (*nativeThread, error) {
+func trapWaitInternal(cctx *proc.ContinueOnceContext, pid int, options trapWaitOptions) (*nativeThread, error) {
+	var waitdbp *nativeProcess = nil
+	if cctx.NumProcs() == 1 {
```

Comment:
I think we should add a comment here because on the surface this looks like a bug, however once you dig deeper you find that `waitdbp.wait()` will behave accordingly if it's receiver is nil.